deepcell_tracking/isbi_utils.py

```diff
@@ -352,23 +370,37 @@ def benchmark_division_performance(trk_gt, trk_res, path_gt, path_res):
     trks = load_trks(trk_res)
     lineage_res, _, y_res = trks['lineages'][0], trks['X'], trks['y']
 
-    # Produce ISBI style text doc to work with
-    trk_to_isbi(lineage_gt, path_gt)
-    trk_to_isbi(lineage_res, path_res)
+    # Produce ISBI style array to work with
+    if path_gt or path_res is not None:
+        print("UserWarning: The 'path_gt` and 'path_res' arguments are deprecated.")
+        trk_to_isbi(lineage_gt, path_gt)
+        trk_to_isbi(lineage_res, path_res)
+    else:
+        gt = trk_to_isbi(lineage_gt)
+        res = trk_to_isbi(lineage_res)
 
     # Match up labels in GT to Results to allow for direct comparisons
     cells_gt, cells_res = match_nodes(y_gt, y_res)
 
     if len(np.unique(cells_res)) < len(np.unique(cells_gt)):
         node_key = {r: g for g, r in zip(cells_gt, cells_res)}
         # node_key maps gt nodes onto resnodes so must be applied to gt
-        G_res = txt_to_graph(path_res, node_key=node_key)
-        G_gt = txt_to_graph(path_gt)
+        if path_gt or path_res is not None:
+            G_res = txt_to_graph(path=path_res, node_key=node_key)
+            G_gt = txt_to_graph(path=path_gt)
+        else:
+            G_res = txt_to_graph(data=res, node_key=node_key)
+            G_gt = txt_to_graph(data=gt)
+
```

Comment:
Could we have a new function: `isbi_to_graph` (or something) that transforms the dataframe into a graph? Maybe we can update `trk_to_isbi` to return the dataframe as well, if that makes things easier.

Then we can update the body of `txt_to_graph` to load the file, then call `isbi_to_graph` and return the result.

And here in the benchmark function we can just call `isbi_to_graph` since we've already warned the user, written any files they care about, and loaded the data. They don't know/care which function we call. This would also help clean up some conditionals.